### PR TITLE
Added crab command

### PR DESCRIPTION
**Pull request by paynejordand**
Can be called with crab, rave, or crab emoji

### CODE REVIEW

**Comment by Jay-Madden** [comment]
Lets move the memes_cog.py file into the memes data folder and rename that folder to just memes_cog. I will also rename the grades_data folder to be grades_cog. This is just clearer and a better way of consolidating concerns

File: bot/cogs/memes_cog/memes_cog.py [hunk 1]
```diff
@@ -6,12 +6,20 @@
 import random
 import time
 
+import io
+
+import os
+import datetime
+from PIL import Image, ImageDraw, ImageSequence, ImageFont
```
Comment (Jay-Madden) [inline bot/cogs/memes_cog/memes_cog.py:13]: When importing from a library its always best to be explicit about what is our code and what is library code, so we shouldnt import types directly instead we should call them with the module name or an alias eg 

`import PIL as p` then `p.Image` or `p.ImageDraw` just to be clear about what our code is calling

File: bot/cogs/memes_cog/memes_cog.py [hunk 2]
```diff
@@ -6,12 +6,20 @@
...
+
 from bot.consts import Colors
 
 log = logging.getLogger(__name__)
 
 max_waldo_grid_size = 100
```
Comment (Jay-Madden) [inline bot/cogs/memes_cog/memes_cog.py:19]: Constants in python are by convention defined as ALL_CAPS_SNAKE_CASE

File: bot/cogs/memes_cog/memes_cog.py [hunk 3]
```diff
@@ -95,5 +103,51 @@ async def spongebob(self, ctx, *, args):
 
         await ctx.send(embed=embed)
 
+    @commands.command(aliases=['rave', '🦀'])
+    async def crab(self, ctx, *, args='Bottom text\n is dead'):
+        # crab.gif dimensions - 352 by 200
```
Comment (Jay-Madden) [inline bot/cogs/memes_cog/memes_cog.py:108]: We need a py doc comment here to have something to show in the !help command when people run that. just a short description of the what and how

File: bot/cogs/memes_cog/memes_cog.py [hunk 4]
```diff
@@ -95,5 +103,51 @@ async def spongebob(self, ctx, *, args):
...
+
+            b = io.BytesIO()
+            frame.save(b, format='GIF')
+            frame = Image.open(b)
+            frames.append(frame)
+
+        # Save, send, and delete created gif
+        frames[0].save(f'bot/cogs/memes_cog/assets/out_{timestamp}.gif', save_all=True, append_images=frames[1:])
+        attachment = discord.File(filename=f'out_{timestamp}.gif', fp=f'bot/cogs/memes_cog/assets/out_{timestamp}.gif')
+        msg = await ctx.send('Generating your gif')
```
Comment (Jay-Madden) [inline bot/cogs/memes_cog/memes_cog.py:145]: Lets move this to the top of the command, in my testing the pillow stuff wasnt a super insignificant amount of time

File: bot/cogs/memes_cog/memes_cog.py [hunk 5]
```diff
@@ -95,5 +103,51 @@ async def spongebob(self, ctx, *, args):
...
+        
+        # Draw the text on to each frame of the gif
+        # Gonna be honest I don't quite understand how it works but I got it from the Pillow docs/issues
+        frames = []
+        for frame in ImageSequence.Iterator(im):
+            d = ImageDraw.Draw(frame)
+            w, h = d.textsize(args, fnt)
+            # draws the text on to the frame. Tries to center horizontally and tries to go as close to the bottom as possible
+            d.text((im.size[0]/2 - w/2, im.size[1] - h - (5 * lines_in_text)), args, font=fnt, align='center', stroke_width=1, stroke_fill=Colors.ClemsonOrange, spacing=6)
+            del d
```
Comment (Jay-Madden) [inline bot/cogs/memes_cog/memes_cog.py:135]: Whats the purpose of del here? is it a pillow thing?

File: bot/cogs/memes_cog/memes_cog.py [hunk 6]
```diff
@@ -95,5 +103,51 @@ async def spongebob(self, ctx, *, args):
...
+            lines_in_text += 1
+        
+        # Draw the text on to each frame of the gif
+        # Gonna be honest I don't quite understand how it works but I got it from the Pillow docs/issues
+        frames = []
+        for frame in ImageSequence.Iterator(im):
+            d = ImageDraw.Draw(frame)
+            w, h = d.textsize(args, fnt)
+            # draws the text on to the frame. Tries to center horizontally and tries to go as close to the bottom as possible
+            d.text((im.size[0]/2 - w/2, im.size[1] - h - (5 * lines_in_text)), args, font=fnt, align='center', stroke_width=1, stroke_fill=Colors.ClemsonOrange, spacing=6)
```
Comment (Jay-Madden) [inline bot/cogs/memes_cog/memes_cog.py:134]: this is clever! however python allows for newlines after commas, we should use that to trim the length of this line down to something manageable

File: bot/cogs/memes_cog/memes_cog.py (hunk 5 above)
Comment (paynejordand) [reply bot/cogs/memes_cog/memes_cog.py:135]: It's a pillow thing. This is how they said to do it

Comment (Jay-Madden) [reply bot/cogs/memes_cog/memes_cog.py:145]: noted

**Review by Jay-Madden — APPROVED** [review]
lgtm

**Pull request merged**